File: project-java-cosminpac04/motorcycleContest/src/main/java/motorcycle/service/TeamService.java
package motorcycle.service;
import java.util.List;
import motorcycle.model.Team;
import motorcycle.repository.TeamRepository;

public class TeamService {
    private final TeamRepository teamRepository;

    public TeamService(TeamRepository teamRepository) {
        this.teamRepository = teamRepository;
    }

    public void addTeam(Team team) {
        teamRepository.add(team);
    }

    public void removeTeam(Team team) {
        teamRepository.delete(team);
    }

    public void updateTeam(Team team, int id){
        teamRepository.update(team, id);
    }

    public Team findTeamById(int id){
        return teamRepository.findById(id);
    }

    public List<Team> findAllTeams(){
        return teamRepository.findAll();
    }

}

File: project-c-cosminpac04/motorcycleApp/Services/TeamService.cs
using System.Collections.Generic;
using motorcycleApp.Models;
using motorcycleApp.Repositories;

namespace motorcycleApp.Services
{
    public class TeamService
    {
        private readonly TeamRepository _teamRepository;

        public TeamService(TeamRepository teamRepository)
        {
            _teamRepository = teamRepository;
        }

        public void AddTeam(Team team)
        {
            _teamRepository.Add(team);
        }

        public void RemoveTeam(int id)
        {
            var team = _teamRepository.FindById(id);
            _teamRepository.Delete(team);
        }

        public void UpdateTeam(int id, Team team)
        {
            _teamRepository.Update(team);
        }

        public Team GetTeamById(int id)
        {
            return _teamRepository.FindById(id);
        }

        public IEnumerable<Team> GetAllTeams()
        {
            return _teamRepository.GetAll();
        }
    }
}